### PR TITLE
Feature: create sign up thunk

### LINKED ISSUE #125: Create Sign Up Action
**__Task Description__**
Create a redux thunk action that can sign up a user by making a call to the appropriate API endpoint.

**__Technical Considerations__**
Should use the `signUp` user service

### PR DESCRIPTION
**Pull request by jaceksupernak**
### Feature: create sign up thunk

**Done:**
- create new branch for this smaller feature, as the last branch after huge rebase was hard to read for new changes
- create signup thunk that at the moment instead of using the service receives data from the mock object
- reafactor Signup component to redirect to the chats page on successfull registration
- refactor Chats data to not throw an error when the friends/chats does not exist yet for fresly registered user
- refactor login reducer to accept email and token

**To-do**
- establish connection with the service once the backend router is merged, which means simply uncommenting the code that is already created
- support error handling

closes #125

### CODE REVIEW

File: client/src/store/features/users/usersSlice.js [hunk 1]
```diff
@@ -46,6 +23,8 @@ export const usersSlice = createSlice({
       ...state,
       id: action.payload.id,
       username: action.payload.username,
+      email: action.payload.username,
```
Comment (moody2times) [inline client/src/store/features/users/usersSlice.js]: Shouldn't this be `email: action.payload.email`?

**Review by moody2times — CHANGES_REQUESTED** [review]
Just confirm for me before I merge this in.

File: client/src/store/features/users/usersSlice.js (hunk 1 above)
Comment (jaceksupernak) [reply client/src/store/features/users/usersSlice.js]: Yes it should! Fixed

**Review by moody2times — APPROVED** [review]
Ok, looking good. 👍🏾

**Pull request merged**